packages/preset-umi/src/features/ssr/webpack/webpack.ts

```diff
@@ -11,9 +11,9 @@ export const build = async (api: IApi, opts: any) => {
   const now = new Date().getTime();
   const bundlerOpts: any = lodash.cloneDeep(opts);
   const oChainWebpack = bundlerOpts.chainWebpack;
-  const isDev = opts.env === Env.development;
+  const isDev = api.env === Env.development;
   const { userConfig, config } = opts;
-  const useHash = (config.hash || (userConfig?.hash && !isDev));
+  const useHash = (config.hash || userConfig?.hash) && !isDev;
```

Comment:
```suggestion
  const useHash = api.config.hash && api.env === Env.production;
```

感觉应该是这样，

1、api.config 里获取的就是考虑了 userConfig 的最终配置
2、判断 production 可能更合理，应该还会有 test 之类的其他环境